File: Python/Easy/1_Two_Sum/solution.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        map = {}

        for i in range(len(nums)):
            complement = target - nums[i]
            if complement in map:
                return [map[complement], i]
            else:
                map[nums[i]] = i
        

File: Golang/Easy/1_Two_Sum/solution.go
package twosum

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int)

	for i, num := range nums {
		complement := target - num
		if index, exist := numMap[complement]; exist {
			return []int{index, i}
		} else {
			numMap[num] = i
		}
	}
	return []int{}
}
